.github/workflows/ci.yml

```diff
@@ -0,0 +1,44 @@
+name: ci
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+      matrix:
+        deno: [v1.x]
+        os:
+          - ubuntu-latest
+          - windows-latest
+          - macOS-latest
+
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v3
+
+      - name: Set up Deno
+        uses: denoland/setup-deno@v1
+        with:
+          deno-version: ${{ matrix.deno }}
+
+      - name: Run checks
+        run: deno task ok
+
+      - name: Run tests
+        run: deno task test
+
+      - name: Generate lcov
+        run: deno task lcov
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v4
+        if: matrix.os == 'ubuntu-latest' && matrix.deno == 'v1.x'
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
We should document how to get this secret and where its used. At the start of that doc we can describe that you need to be part of a special permissions group to do it etc etc

Reply:
Done!